db/migrate/20170304172450_create_player_selections.rb

```diff
@@ -0,0 +1,11 @@
+class CreatePlayerSelections < ActiveRecord::Migration[5.0]
+  def change
+    create_table :player_selections do |t|
+      t.references :player, null: false
+      t.references :hero, null: false
+      t.string :role
+      t.references :composition, null: false
+      t.timestamps
+    end
```

Comment:
Ooh this is a good point! So initially we didn't have ~hero\_pools~ _player\_heroes_ in the data schema, it came about later after https://trello.com/c/J2nzZThf/6-feature-to-show-hero-pool-for-each-player was proposed. Your point about renaming the hero\_pools table makes sense and, from that rename, it's easy to see that individually referencing hero\_id and player\_id here in player\_selections could instead be done by referencing player\_hero\_id.

I do like the implications of the team organizer being able to see a list of player-hero combinations with a high `confidence` value, so when the organizer thinks "I need a healer," the first players suggested for that role are those known to be good healers. I think using player\_hero\_id in this table doesn't have to be limiting, either: the team organizer can still put whichever player on whichever hero and we'll just create a new player\_heroes record with 0 `confidence`. This feels necessary because not every player will necessarily be using this app. Maybe it's only the organizer using the app, and their teammates aren't around to specify their mains in the player\_heroes table.